pilot/pkg/xds/debug.go

```diff
@@ -687,11 +688,20 @@ func (s *DiscoveryServer) ConfigDump(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 	if ts := s.getResourceTypes(req); len(ts) != 0 {
+		if con.proxy.IsZTunnel() {
+			istiolog.Warnf("proxy %v is not envoy based", proxyID)
```

Comment:
do we need this?

Reply:
was curious about this case -  requesting things like listeners/cluster/routes/etc through `types` query params are not going to apply to ztunnels and could be confusing to users if they're expecting them to work. WDYT?

Reply:
This is a low level debugging interface, IMO we should just return what they ask. Users aren't going to see the istiod logs anyways, and its perfectly valid for ztunnel's to call this (and not put types as listners, etc)

Reply:
yeah, not sure why users would want to see these ztunnel infos.